matrix_sdk_base/src/models/room.rs

```diff
@@ -617,14 +633,36 @@ impl Room {
     /// Returns true if `MessageQueue` was added to.
     #[cfg(feature = "messages")]
     #[cfg_attr(docsrs, doc(cfg(feature = "messages")))]
-    pub fn handle_message(&mut self, event: &MessageEvent) -> bool {
-        self.messages.push(event.clone())
+    pub fn handle_message(&mut self, event: &AnyMessageEventStub) -> bool {
+        let message = MessageWrapper::clone_into_any_content(event);
+        self.messages.push(message)
+    }
+
+    /// Handle a room.redaction event and update the `MessageQueue` if necessary.
+    ///
+    /// Returns true if `MessageQueue` was updated.
+    #[cfg(feature = "messages")]
+    #[cfg_attr(docsrs, doc(cfg(feature = "messages")))]
+    pub fn handle_redaction(&mut self, event: &RedactionEventStub) -> bool {
+        if let Some(msg) = self
+            .messages
+            .iter_mut()
+            .find(|msg| event.redacts == msg.event_id)
+        {
+            msg.content = AnyMessageEventContent::RoomRedaction(event.content.clone());
+
+            let redaction = full_event_from_stub(event.clone(), self.room_id.clone());
+            msg.unsigned.redacted_because = Some(EventJson::from(redaction));
```

Comment:
Good thinking, I forgot that we'll want the reason available when we restore the event from the store.